test/model/tools/model_test/main.cpp

```diff
@@ -0,0 +1,379 @@
+                update_spec(spec, conn_config, table_config, __wt_optarg);
+                break;
+            case 'h':
+                home = __wt_optarg;
+                break;
+            case 'I':
+                min_iterations = parse_uint64(__wt_optarg);
+                break;
+            case 'i':
+                load_spec(spec, conn_config, table_config, __wt_optarg);
+                break;
+            case 'l':
+                p = parse_uint64_range(__wt_optarg);
+                spec.min_sequences = p.first;
+                spec.max_sequences = p.second;
+                if (p.first <= 0)
+                    throw std::runtime_error("Not enough transactions");
+                break;
+            case 'M':
+                p = parse_uint64_range(__wt_optarg);
+                spec.min_tables = p.first;
+                spec.max_tables = p.second;
+                if (p.first <= 0)
+                    throw std::runtime_error("Not enough tables");
+                break;
+            case 'n':
+                print_only = true;
+                break;
+            case 'p':
+                preserve = true;
+                break;
+            case 'S':
+                base_seed = parse_uint64(__wt_optarg);
+                break;
+            case 'T':
+                table_config += ",";
+                table_config += __wt_optarg;
+                break;
+            case 't':
+                min_runtime_s = parse_uint64(__wt_optarg);
+                break;
+            case '?':
+                usage(progname);
+                return EXIT_SUCCESS;
+            default:
+                usage(progname);
+                return EXIT_FAILURE;
+            }
+        argc -= __wt_optind;
+        if (argc != 0) {
+            usage(progname);
+            return EXIT_FAILURE;
+        }
+    } catch (std::exception &e) {
+        std::cerr << progname << ": " << e.what() << std::endl;
+        return EXIT_FAILURE;
+    }
+
+    /* Run the test, potentially many times. */
+    double start_time = current_time();
+    for (uint64_t iteration = 1;; iteration++) {
+        uint64_t seed = base_seed + iteration;
+        std::cout << "Iteration " << iteration << ", seed 0x" << std::hex << seed << std::dec
+                  << std::endl;
+
+        /* Generate the workload. */
+        std::shared_ptr<model::kv_workload> workload;
+        try {
+            workload = model::kv_workload_generator::generate(spec, seed);
+        } catch (std::exception &e) {
+            std::cerr << "Failed to generate the workload: " << e.what() << std::endl;
+            return EXIT_FAILURE;
+        }
+
+        /* If we only want to print the workload, then do so. */
+        if (print_only) {
+            std::cout << *workload.get();
+            break;
+        }
+
+        /* Clean up the previous database directory, if it exists. */
+        testutil_remove(home.c_str());
+
+        /* Run and verify the workload. */
```

Comment:
This comment can go IMO - it's not adding much :-)

Reply:
This is a matter of style. In terms of explaining the code below, it indeed does not add much. The reason why it is there is because it acts as a "title" for the code below, so that if you just read those "titles", you will see everything that is happening in the code block without even reading the code.

Reply:
Fair enough. I have a pathological thing against comments that repeat the code but you're right that it's a style thing. Feel free to leave it.